projects/packages/jetpack-mu-wpcom/src/features/block-theme-previews/modal.jsx

```diff
@@ -0,0 +1,82 @@
+import { Button, Modal } from '@wordpress/components';
+import { useDispatch, useSelect } from '@wordpress/data';
+import { useEffect } from '@wordpress/element';
+import { __, sprintf } from '@wordpress/i18n';
+import React from 'react';
+import { getPreviewedThemeStylesheet } from './utils';
+
+import './modal.scss';
+
+/**
+ * The modal that shows in the Site Editor the first time a user previews a block theme.
+ */
+export default function BlockThemePreviewsModal() {
+	const stylesheet = getPreviewedThemeStylesheet();
+	const theme = useSelect( select => select( 'core' ).getTheme( stylesheet ), [ stylesheet ] );
+
+	const isInSiteEditor = useSelect( select => !! select( 'core/edit-site' ), [] );
+	const isModalVisible = useSelect(
+		// eslint-disable-next-line @typescript-eslint/no-explicit-any
+		select => select( 'automattic/wpcom-block-theme-previews' ).isModalVisible(),
+		[]
+	);
+	const { dismissModal } = useDispatch( 'automattic/wpcom-block-theme-previews' );
+	const { suppressLivePreviewModal } = useDispatch( 'automattic/wpcom-welcome-guide' );
+
+	useEffect( () => {
+		if ( isInSiteEditor ) {
+			// Hide the legacy modal from the WordPress Editing Toolkit plugin.
+			suppressLivePreviewModal?.();
+		}
+	}, [ isInSiteEditor, suppressLivePreviewModal ] );
+
+	if ( ! isInSiteEditor || ! isModalVisible ) {
+		return null;
+	}
+	return (
+		<Modal
+			className="wpcom-block-theme-previews-modal"
+			onRequestClose={ dismissModal }
+			shouldCloseOnClickOutside={ false }
+		>
+			<div className="wpcom-block-theme-previews-modal__content">
+				<div className="wpcom-block-theme-previews-modal__text">
+					<h1 className="wpcom-block-theme-previews-modal__heading">
+						{ sprintf(
+							// translators: %s: theme name
+							__( 'You’re previewing %s', 'jetpack-mu-wpcom' ),
```

Comment:
Should we make this `You’re previewing the %s theme'`?
For example, when we try out the Organizer theme, it would be nice if this notice would say, “Previewing the Organizer Theme." The current implementation says “Previewing Organizer”, which might cause misunderstanding. pekYwv-3ut-p2

```suggestion
							__( 'You’re previewing the %s theme', 'jetpack-mu-wpcom' ),
```

Reply:
We can, but the heading could potentially span three lines for long theme name, e.g.

<img width="897" alt="image" src="https://github.com/Automattic/jetpack/assets/1525580/bae42c53-759f-4afb-8c98-f20e7b58b77b">

I have no preferences. What do you think @autumnfjeld @matt-west ?

Reply:
I deliberately shortened the copy to avoid title from spanning three lines (in most cases). Given that the user is seeing this after coming from the theme details page, I don't think we need to reiterate that it's a theme.

Reply:
Cool, we'll keep the copy as-is for now, thanks!